### DISCUSSION TITLE
请教下3.9如何获取图片？

### DISCUSSION THREAD

**Discussion by wuwei0509** [ttttupup/wxhelper#201] [Q&A] [answered]
看了下3.9：
1、image_hook没有实现。
2、有一个实现是WECHAT_GET_IMG_BY_NAME，参数是imagePath 和 savePath
      其中的imagePath如何获得？

从msg hook里获得的dat形如：
> 'path': 'wxid_xxxxx\\FileStorage\\MsgAttach\\2cb11c0d73f8459484xxxxxxx\\Image\\2023-07\\ddb88e70e6537c268b5fa1778eb2d3ba.dat'

最为imagePath使用后，没有作用

**Comment by ttttupup** [comment, marked as answer]
图片路径，hook的消息里有后缀，和个人信息里的返回的data
path拼接就是完整的路径。解密接口需要完整的路径。

**Comment by wuwei0509** [comment]
感谢，可以了。
图片源地址为：
个人信息的"dataSavePath"+ hook返回的"path"

**Comment by Thick-z** [comment]
3.9.5.81里没有datasavepath,hook到的base64Img是一个超低像素的缩略图，如何获取到原图？

**Comment by chengxiangtong** [comment]
3.9.8.25图片并不会自动下载，收到的消息里也没有图片的ID，请问大家怎么获取图片？
